llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -800,6 +800,13 @@ instCombineConvertFromSVBool(InstCombiner &IC, IntrinsicInst &II) {
 
 static std::optional<Instruction *> instCombineSVESel(InstCombiner &IC,
                                                       IntrinsicInst &II) {
+  // svsel(ptrue, x, y) => x
+  auto *OpPredicate = II.getOperand(0);
+  if (match(OpPredicate, m_Intrinsic<Intrinsic::aarch64_sve_ptrue>(
+                             m_ConstantInt<AArch64SVEPredPattern::all>()))) {
```

Comment:
What about using `isAllActivePredicate` here to catch more cases?

Reply:
Thanks @paulwalker-arm , update with API `isAllActivePredicate `